### DISCUSSION TITLE
Pipeline fails with cypress.config.js as ES module

### DISCUSSION THREAD

**Discussion by kallgren** [DevSkillsHQ/discusssions#42] [Help] [answered]
Hey there,

I did the frontend test using SvelteKit, but I am now running into problems at the **Fetch apiUrl** step in the pipeline. SvelteKit uses ES modules (`"type": "module"` in package.json), so I had to rewrite cypress.config.js to use ES syntax instead of CommonJS. However, the **Fetch apiUrl** step tries to load cypress.config.js as a CommonJS module, using `require()`, which now fails.

Any suggestions for how I can solve this? The suggested method in the screenshot of renaming to cypress.config.cjs won't work since the workflow explicitly looks for cypress.config.js, and changing all other files to .mjs is messy (and perhaps not even possible with .svelte files?). Is there perhaps any way to change the pipeline configuration to use `import` instead?

Thanks!

<img width="712" alt="image" src="https://github.com/DevSkillsHQ/discusssions/assets/5982001/41fd329c-e0e3-4148-b26d-6b9e8aa3cc9e">

**Comment by fenske** [comment]
Hi @kallgren. I replied in [this thread](https://github.com/ffc1e12/transaction-management-frontend-level-3_12027bd-e1j4jc/pull/1#issuecomment-1736244901). Let's continue there.

**Comment by kallgren** [comment, marked as answer]
@fenske solved it simply by renaming the config to **cypress.config.ts**. Thanks!

(I also upgraded Cypress to v13+ which may or may not have helped)
